test/tt_store_SUITE.erl

```diff
@@ -0,0 +1,99 @@
+          testExactTime,
+          testIncluded,
+          testIncludedMultipleAll,
+          testIncludedMultipleSome,
+          testOverlappingAtStart,
+          testOverlappingAtEnd,
+          testNotOverlapping,
+          testNameFound,
+          testNameNotFound].
+
+testReversedTimes(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {12, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    Res = tt_store:list(),
+    Res = [].
+
+testList(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    tt_store:add("Second talk",   {{2016, 5, 5}, {11, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}, "Room 2"),
+    Res = tt_store:list(),
+    Res = [{talk, "First talk",  {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"},
+           {talk, "Second talk", {{2016, 5, 5}, {11, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}, "Room 2"}].
+
+testExactTime(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    Res = tt_store:find_by_time(  {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}),
+    Res = [{talk, "First talk",   {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"}].
+
+testIncluded(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    Res = tt_store:find_by_time(  {{2016, 5, 5}, { 8, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}),
+    Res = [{talk, "First talk",   {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"}].
+
+testIncludedMultipleAll(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    tt_store:add("Second talk",   {{2016, 5, 5}, {11, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}, "Room 2"),
+    Res = tt_store:find_by_time(  {{2016, 5, 5}, { 8, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}),
+    Res = [{talk, "First talk",   {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"},
+           {talk, "Second talk",  {{2016, 5, 5}, {11, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}, "Room 2"}].
+
+testIncludedMultipleSome(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    tt_store:add("Second talk",   {{2016, 5, 5}, {11, 00, 0}}, {{2016, 5, 5}, {12, 00, 0}}, "Room 2"),
+    Res = tt_store:find_by_time(  {{2016, 5, 5}, { 8, 00, 0}}, {{2016, 5, 5}, {11, 30, 0}}),
+    Res = [{talk, "First talk",   {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"}].
+
+testOverlappingAtStart(_) ->
+    tt_store:start_link(),
+    tt_store:add("First talk",    {{2016, 5, 5}, {10, 00, 0}}, {{2016, 5, 5}, {11, 00, 0}}, "Room 1"),
+    Res = tt_store:find_by_time(  {{2016, 5, 5}, { 9, 00, 0}}, {{2016, 5, 5}, {10, 30, 0}}),
+    Res = [].
```

Comment:
Now I think that it would be better having the `First talk` in the result for `testOverlappingAtStart` since we would like to post talks that "will start in a given time range". 

In face of that we can think of the call 

```erlang
tt_store:find_by_time(  {{2016, 5, 5}, { 9, 00, 0}}, {{2016, 5, 5}, {10, 30, 0}})
```

as "give me all talks that start between 9:00 and 10:30 on the given day".